src/Tools/AI Test Toolkit/src/TestSuite/AITTestSuiteImportExport.XmlPort.al

```diff
@@ -21,6 +21,37 @@ xmlport 149031 "AIT Test Suite Import/Export"
                 fieldattribute(Code; AITSuite.Code)
                 {
                     Occurrence = Required;
+
+                    trigger OnAfterAssignField()
+                    var
+                        AITTestSuiteRec: Record "AIT Test Suite";
+                        AITTestSuiteMgt: Codeunit "AIT Test Suite Mgt.";
+                        CallerModuleInfo: ModuleInfo;
+                        SameSuiteDifferentXMLErr: Label 'The test suite %1 is already imported with a different XML by the same app. Please delete the test suite and import again.', Comment = '%1 = Test Suite Code';
+                        SameSuiteDifferentAppErr: Label 'The test suite %1 is already imported by a different app. Please rename the test suite and import again.', Comment = '%1 = Test Suite Code';
+                    begin
+                        // Skip if the same suite is already imported by the same app
+                        // Error if the same suite is already imported with a different XML
+                        // Error if the same suite is already imported by a different app
+                        AITTestSuiteMgt.GetCallerModuleInfo(CallerModuleInfo);
+                        AITSuite."Imported by AppId" := CallerModuleInfo.Id;
+
+                        AITSuite."Imported XML's MD5" := MD5FileHash;
+
+                        AITTestSuiteRec.SetLoadFields(Code, "Imported by AppId", "Imported XML's MD5");
+                        AITTestSuiteRec.SetRange(Code, AITSuite.Code);
+
+                        if AITTestSuiteRec.FindFirst() then
+                            if AITTestSuiteRec."Imported by AppId" = CallerModuleInfo.Id then
+                                if AITTestSuiteRec."Imported XML's MD5" = AITSuite."Imported XML's MD5" then begin
+                                    SkipTestSuites.Add(AITSuite.Code);
+                                    currXMLport.Skip();
```

Comment:
There is also a keyword to stop the execution of the XML port. Maybe you want this one?

Reply:
Don't want to quite the report yet, as the xml can have multiple suite:

For example:

```
<?xml version="1.0" encoding="UTF-16" standalone="no"?>
<Root>
  <AITSuite Code="ACCURACY" Description="Accuracy" Dataset="ACCURACY_EXTRACTINFOFROMCSVPROMPT">
    <Line CodeunitID="149826" Dataset="ACCURACY_EXTRACTINFOFROMCSVPROMPT" />
  </AITSuite>
  <AITSuite Code="FUNC" Description="Functional Tests" Dataset="EXTRACTINFOFROMCSVPROMPT">
    <Line CodeunitID="149820" Dataset="EXTRACTINFOFROMCSVPROMPT" />
    <Line CodeunitID="149821" Dataset="MAGICFUNCTIONATTACHMENTPROMPT" />
    <Line CodeunitID="149822" Dataset="LOADMAPPINGFROMCSV" />
  </AITSuite>
</Root>
```